### PR TITLE
GMTDataArrayAccessor: Add inline examples for setting GMT specific properties (#2370)Co-authored-by: Michael Grund <23025878+michaelgrund@users.noreply.github.com>

### PR DESCRIPTION
**Pull request by seisman**
**Description of proposed changes**

Fixes #

**Reminders**

- [ ] Run `make format` and `make check` to make sure the code follows the style guide.
- [ ] Add tests for new features or tests that would have caught the bug that you're fixing.
- [ ] Add new public functions/methods/classes to `doc/api/index.rst`.
- [ ] Write detailed docstrings for all functions/methods.
- [ ] If wrapping a new module, open a 'Wrap new GMT module' issue and submit reasonably-sized PRs.
- [ ] If adding new functionality, add an example to docstrings or tutorials.
- [ ] Use underscores (not hyphens) in names of Python files and directories.

**Slash Commands**

You can write slash commands (`/command`) in the first line of a comment to perform
specific operations. Supported slash commands are:

- `/format`: automatically format and lint the code
- `/test-gmt-dev`: run full tests on the latest GMT development version

### CODE REVIEW

File: pygmt/accessors.py [hunk 1]
```diff
@@ -9,20 +9,43 @@
...
     >>> # See if grid uses Cartesian (0) or Geographic (1) coordinate system
     >>> grid.gmt.gtype
     1
+
+    You can also set the GMT specific properties for grids created by yourself:
+
+    >>> import numpy as np
+    >>> import pygmt
+    >>> import xarray as xr
+    >>> # creata a DataArray in gridline coordinates of sin(lon) * cos(lat)
```
Comment (michaelgrund) [inline pygmt/accessors.py:37]:
```suggestion
    >>> # create a DataArray in gridline coordinates of sin(lon) * cos(lat)
```

**Review by michaelgrund — APPROVED** [review]

File: pygmt/accessors.py [hunk 2]
```diff
@@ -9,20 +9,43 @@
...
     >>> # See if grid uses Gridline (0) or Pixel (1) registration
     >>> grid.gmt.registration
     1
     >>> # See if grid uses Cartesian (0) or Geographic (1) coordinate system
     >>> grid.gmt.gtype
     1
+
+    You can also set the GMT specific properties for grids created by yourself:
+
+    >>> import numpy as np
```
Comment (seisman) [inline pygmt/accessors.py:34]: Note: this inline example is modified from https://github.com/GenericMappingTools/pygmt/issues/390.

Comment (michaelgrund) [reply pygmt/accessors.py:34]: ok

**Comment by seisman** [comment]
I plan to add more inline examples after addressing some related issues, so mark it as WIP.

**Comment by seisman** [comment]
> I plan to add more inline examples after addressing some related issues, so mark it as WIP.

Changed my mind. I'll add more example when addresing accessor-related issues. Marking this PR as "final review call".

**Review by willschlitzer — APPROVED** [review]
Looks good!

**Pull request merged**